kitsune/community/api.py

```diff
@@ -0,0 +1,334 @@
+        combined = defaultdict(lambda: {
+            'answer_count': 0,
+            'solution_count': 0,
+            'helpful_vote_count': 0,
+        })
+
+        for d in creator_answer_counts:
+            combined[d['term']]['user_id'] = d['term']
+            combined[d['term']]['answer_count'] = d['count']
+
+        for d in creator_solutions_counts:
+            combined[d['term']]['user_id'] = d['term']
+            combined[d['term']]['solution_count'] = d['count']
+
+        for d in creator_helpful_counts:
+            combined[d['term']]['user_id'] = d['term']
+            # Since this is a term_stats filter, not just a term filter, it is total, not count.
+            combined[d['term']]['helpful_vote_count'] = int(d['total'])
+
+        # Sort by answer count, and get just the ids into a list.
+        sort_key = self.query_values['ordering']
+        if sort_key[0] == '-':
+            sort_reverse = True
+            sort_key = sort_key[1:]
+        else:
+            sort_reverse = False
+
+        top_contributors = combined.values()
+        top_contributors.sort(key=lambda d: d[sort_key], reverse=sort_reverse)
+        user_ids = [c['user_id'] for c in top_contributors]
+        full_count = len(user_ids)
+
+        # Paginate those user ids.
+        try:
+            page = int(self.request.GET.get('page', 1))
+        except ValueError:
+            page = 1
+        count = 10
+        page_start = (page - 1) * count
+        page_end = page_start + count
+        user_ids = user_ids[page_start:page_end]
+
+        # Get full user objects for every id on this page.
+        users = UserMappingType.reshape(
+            UserMappingType
+            .search()
+            .filter(id__in=user_ids)
+            .values_dict('id', 'username', 'display_name', 'avatar', 'last_contribution_date')
+            [:count])
+
+        # For ever user object found, mix in the metrics counts for that user,
+        # and then reshape the data to make more sense to clients.
+        data = []
+        for u in users:
+            d = combined[u['id']]
+            d['user'] = u
+            d['last_contribution_date'] = d['user'].get('last_contribution_date', None)
+            d.pop('user_id', None)
+            d['user'].pop('id', None)
+            d['user'].pop('last_contribution_date', None)
+            data.append(d)
+
+        # One last sort, since ES didn't return the users in any particular order.
+        data.sort(key=lambda d: d[sort_key], reverse=sort_reverse)
+
+        # Add ranks to the objects.
+        for i, contributor in enumerate(data, 1):
+            contributor['rank'] = page_start + i
+
+        return {
+            'results': data,
+            'count': full_count,
+            'filters': self.query_values,
+            'allowed_orderings': [
+                'answer_count',
+                'solution_count',
+                'helpful_vote_count',
```

Comment:
So, I'm not entirely sure I understand the question. Why can't you do another ES search ordered by `last_contribution_date` and just retrieve that page of ids?

Generally, SUMO bugs tend to be under-specified, so I'd treat everything as a "first pass", land it and then create new bugs for optimization work or changes in requirements. The theory being that they need to see a prototype to help them figure out what it is they need and spending the least amount of time on the prototype gets it out there faster and "wastes" less time.

Going on that theory, I'd just leave it as is for now and push off additional work to new bugs.